libs/utils/benchmark/benchmark_memcpy.cpp

```diff
@@ -18,25 +18,127 @@
 
 #include <benchmark/benchmark.h>
 
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <vector>
+
+// ------------------------------------------------------------------------------------------------
+
+#ifdef __linux__
+#include <sched.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#endif
+
+// Returns true if successful, false if failed
+static bool pinThreadToCore(int core_id) {
+#ifdef __linux__
+    cpu_set_t cpuset;
+    CPU_ZERO(&cpuset);
+    CPU_SET(core_id, &cpuset);
+
+    // 0 = current thread
+    int const result = sched_setaffinity(0, sizeof(cpu_set_t), &cpuset);
+    return (result == 0);
+#else
+    (void)core_id;
+    return -1;
+#endif
+}
+
+// ------------------------------------------------------------------------------------------------
+
+static constexpr size_t ALIGNMENT = 16 << 10;
+
+template<class T, size_t Align>
+struct AlignedAllocator {
+    using value_type = T;
+
+    template<class U>
+    struct rebind {
+        using other = AlignedAllocator<U, Align>;
+    };
```

Comment:
unused?

Reply:
yeah, but needed, doesn't compile without it.